### PR TITLE
Add boost include dirs explicitly

### PR DESCRIPTION
**Pull request by gvardany**
**Description of work.**

Add boost include directories explicitly as system headers to the relevant targets.
Clang 10 does not compile without this.

**To test:**

Code review.

*There is no associated issue, no release notes, internal change.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/CMakeLists.txt [hunk 1]
```diff
@@ -10,8 +10,7 @@ set(Boost_LIBRARIES ${Boost_LIBRARIES} ${old_boost})
 
 # NumPy
 find_package(Numpy REQUIRED)
-include_directories(SYSTEM ${PYTHON_INCLUDE_PATH})
-include_directories(SYSTEM ${PYTHON_NUMPY_INCLUDE_DIR})
+include_directories(SYSTEM ${PYTHON_INCLUDE_PATH} ${PYTHON_NUMPY_INCLUDE_DIR} ${Boost_INCLUDE_DIRS})
```
Comment (quantumsteve) [inline Framework/PythonInterface/CMakeLists.txt]: Is it worth switching these to `target_include_directories`?

File: Framework/MDAlgorithms/CMakeLists.txt [hunk 2]
```diff
@@ -426,7 +426,7 @@ endif()
 # Add to the 'Framework' group in VS
 set_property(TARGET MDAlgorithms PROPERTY FOLDER "MantidFramework")
 
-include_directories(inc)
+include_directories(inc SYSTEM ${Boost_INCLUDE_DIRS})
```
Comment (quantumsteve) [inline Framework/MDAlgorithms/CMakeLists.txt]: Is it worth switching these to `target_include_directories`?

File: Framework/Geometry/CMakeLists.txt [hunk 3]
```diff
@@ -453,12 +453,12 @@ if(UNITY_BUILD)
...
 
 # A few defines needed for OpenCascade on the Mac
+
 if(APPLE)
   add_definitions(-DHAVE_IOSTREAM -DHAVE_LIMITS -DHAVE_IOMANIP)
 endif()
 
-include_directories(${OPENGL_INCLUDE_DIR} ${TBB_INCLUDE_DIRS})
-include_directories(${GSL_INCLUDE_DIR})
+include_directories (SYSTEM ${OPENGL_INCLUDE_DIR} ${TBB_INCLUDE_DIRS} ${GSL_INCLUDE_DIR} ${Boost_INCLUDE_DIRS})
```
Comment (quantumsteve) [inline Framework/Geometry/CMakeLists.txt]: Is it worth switching these to `target_include_directories`?

**Review by quantumsteve — CHANGES_REQUESTED** [review]
Explicitly including the Boost headers looks like the right thing to do, plus could be helpful if Boost was installed in a nonstandard location.

Should we take this opportunity to move `include_directories` to `target_include_directories`? Please respond if any of the comments would require substantial changes to implement.

File: Framework/PythonInterface/CMakeLists.txt (hunk 1 above)
Comment (gvardany) [reply Framework/PythonInterface/CMakeLists.txt]: I think it's more complicated for PythonInterface; it is not a library but a custom target, I am not sure how to make this work with `target_include_directories`.

**Review by quantumsteve — APPROVED** [review]

**Pull request merged**